docs/cip/README.md

```diff
@@ -461,247 +1988,671 @@ The Leios protocol may have to mitigate the following categories of threats.
+
+The following plot shows a forecast for SPO profitability under Leios, assuming
+a "business as usual" scenario where the fee, treasury, and monetary expansion
+protocol parameter stay the same as presently. The precise profitability of
+individual SPOs depends strongly upon how they host their nodes, but there is a
+clear trend towards profitability (without any contributions from the Cardano
+Reserve) once 30-50 transactions per second are sustained. Note that
+profitability slows at very high throughput because of the substantial expense
+of network egress and storage of the ledger.
 
-| Year | Reserve Depletion | Rewards from Fees (ADA) | Required TPS (Avg Tx) | Required IB/s |
-| ---: | ----------------: | ----------------------: | --------------------: | ------------: |
-| 2025 |               ~0% |                       0 |                     0 |             0 |
-| 2026 |              ~13% |               6,240,000 |                  10.9 |          0.16 |
-| 2027 |              ~24% |              11,520,000 |                  20.1 |          0.29 |
-| 2028 |              ~34% |              16,320,000 |                  28.5 |          0.41 |
-| 2029 |              ~43% |              20,640,000 |                  36.1 |          0.52 |
-| 2030 |              ~50% |              24,000,000 |                  41.9 |          0.60 |
+![SPO profitability under Leios.](images/leios-forecast-sqrt-fill.svg)
 
-Note that by 2029, to compensate for Reserve depletion, the network would need to process approximately 36 TPS with average-sized transactions, requiring an Input Block rate of around 0.52 IB/s, roughly 10 times the current mainnet throughput. Leios's design would comfortably support this increased throughput while maintaining decentralization.
+<em>Figure 17: SPO profitability under Leios</em>
+
+<a id="intensive-plutus-execution" href="#intensive-plutus-execution">**Intensive
+Plutus execution**</a>
+
+Because there typically is a time window of several seconds from the time a
+Leios endorser block can be created to when voting must complete, there is also
+an opportunity to do more computation validating transactions within an endorser
+block than for a Praos ranking block. This opens the possibility of increasing
+the Plutus execution budget for endorser blocks so that it is significantly
+larger than the budget for Praos blocks. At the very least a script could be
+allowed to use the whole Plutus execution budget for an endorser block, instead
+of just one quarter of it as is the case for Praos.
+
+Numerous emerging use cases on Cardano would benefit from larger Plutus
+execution budgets. Complex dapps currently have to split a single logical
+operation into a sequence of several transactions, increasing the development
+effort, the complexity, and the attack surface of the scripts involved.
+
+- _ZK proofs:_ It may be possible to increase the Plutus execution budget enough
```

Comment:
IIRC, it took several scripts indeed to verify a zk-proof of a pairing check onchain (RSNARK team project). Adam should be the best person to contact to know how to do this and more importantly if there is any plan to facilitate this.